lucene/misc/native/build.gradle

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This gets separated out from misc module into a native module due to incompatibility between cpp-library and java-library plugins.
+ * For details, please see https://github.com/gradle/gradle-native/issues/352#issuecomment-461724948
+ */
+import org.apache.tools.ant.taskdefs.condition.Os
+
+description = 'Module for native code'
+
+apply plugin: 'cpp-library'
+
+library {
+  baseName = 'NativePosixUtil'
+
+  // Native build for Windows platform will be added in later stage
+  targetMachines = [
+      machines.linux.x86_64,
+      machines.macOS.x86_64,
+      machines.windows.x86_64
+  ]
+
+  // Point at platform-specific sources. Other platforms will be ignored
+  // (plugin won't find the toolchain).
+  if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+    source.from file("${projectDir}/src/main/windows")
+  } else if (Os.isFamily(Os.FAMILY_UNIX) || Os.isFamily(Os.FAMILY_MAC)) {
+    source.from file("${projectDir}/src/main/posix")
+  }
+}
+
+tasks.withType(CppCompile).configureEach {
+  def javaHome = rootProject.ext.runtimeJava.getInstallationDirectory().getAsFile().getPath()
+
+  // Assume standard openjdk layout. This means only one architecture-specific include folder
+  // is present.
+  systemIncludes.from file("${javaHome}/include")
+
+  for (def path : [
+      file("${javaHome}/include/win32"),
+      file("${javaHome}/include/darwin"),
+      file("${javaHome}/include/linux"),
+      file("${javaHome}/include/solaris")]) {
+    if (path.exists()) {
+      systemIncludes.from path
+    }
+  }
+
+  compilerArgs.add '-fPIC'
```

Comment:
Just curious, shall we also modify the compiler args when it’s on Windows, to be the same with what's used before?

https://github.com/apache/lucene-solr/blob/ec9a659845973a0dd0ee7c04e0075db818ed118d/lucene/misc/src/java/org/apache/lucene/store/WindowsDirectory.java#L31-L35 

A quick search shows that some of these flags might be specific to MinGW compiler though, so I'm not sure if these flags are still relevant.

Reply:
I wouldn't add anything if it works (and it currently does for me). There are compiler toolchain-specific ways of adding options but like I said - if it's not required, I wouldn't worry about it.

Reply:
Ok cool. I've updated the comment here to reflect the new way of building the code.